test/Microsoft.AspNetCore.Mvc.Razor.Precompilation.Tests/ApplicationConsumingPrecompiledViews.cs

```diff
@@ -0,0 +1,120 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Server.IntegrationTesting;
+using Microsoft.DotNet.Cli.Utils;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.Precompilation
+{
+    public class ApplicationConsumingPrecompiledViews 
+        : IClassFixture<ApplicationConsumingPrecompiledViews.ApplicationConsumingPrecompiledViewsFixture>
+    {
+        public ApplicationConsumingPrecompiledViews(ApplicationConsumingPrecompiledViewsFixture fixture)
+        {
+            Fixture = fixture;
+        }
+
+        public ApplicationTestFixture Fixture { get; }
+
+        [Theory]
+        [InlineData(RuntimeFlavor.Clr)]
+        [InlineData(RuntimeFlavor.CoreClr)]
+        public async Task ConsumingClassLibrariesWithPrecompiledViewsWork(RuntimeFlavor flavor)
+        {
+            // Arrange
+            using (var deployer = Fixture.CreateDeployment(flavor))
+            {
+                var deploymentResult = deployer.Deploy();
+                var httpClient = new HttpClient()
+                {
+                    BaseAddress = new Uri(deploymentResult.ApplicationBaseUri)
+                };
+
+                // Act
+                var response = await httpClient.GetStringAsync("Manage/Home");
+
+                // Assert
+                TestEmbeddedResource.AssertContent("ApplicationConsumingPrecompiledViews.Manage.Home.Index.txt", response);
+            }
+        }
+
+        public class ApplicationConsumingPrecompiledViewsFixture : ApplicationTestFixture
+        {
+            private string _packOutputDirectory = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
+
+            public ApplicationConsumingPrecompiledViewsFixture()
+                : base("ApplicationUsingPrecompiledViewClassLibrary")
+            {
+                ClassLibraryPath = Path.GetFullPath(Path.Combine(ApplicationPath, "..", "ClassLibraryWithPrecompiledViews"));
+            }
+
+            private string ClassLibraryPath { get; }
+
+            protected override void Restore()
+            {
+                CreateClassLibraryPackage();
+                RestoreProject(ApplicationPath, new[] { _packOutputDirectory });
+            }
+
+            private void CreateClassLibraryPackage()
+            {
+                RestoreProject(ClassLibraryPath);
+                ExecuteForClassLibrary(Command.CreateDotNet("build", new[] { ClassLibraryPath, "-c", "Release" }));
+                ExecuteForClassLibrary(Command.CreateDotNet(
+                    "razor-precompile", 
+                    GetPrecompileArguments("net451")));
+
+                ExecuteForClassLibrary(Command.CreateDotNet(
```

Comment:
It's not a class library. I guess we could call it portable application but @Eilon does not want us using that term since it's a bit loaded?